src/borg/testsuite/repository.py

```diff
@@ -133,6 +133,7 @@ def test_single_kind_transactions(self):
     def test_list(self):
         for x in range(100):
             self.repository.put(H(x), b'SOMEDATA')
+        self.repository.commit()
```

Comment:
Likely because iter_objects of LoggedIO doesn't have these lines:

```py
        if segment == self.segment and self._write_fd:
            self._write_fd.sync()
```

Therefore the segment header etc. could all still be in an internal buffer, and not in the OS-level file.